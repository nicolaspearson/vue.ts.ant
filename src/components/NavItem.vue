<template>
  <section class="nav-item" :class="navItem.disabled ? 'disabled' : navItem.color">
    <router-link class="link" :to="navItem.location">
      <div class="text">
        <span>{{ navItem.text }}</span>
      </div>
    </router-link>
  </section>
</template>

<script lang="ts" >
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component({})
export default class NavItem extends Vue {
  @Prop({ required: true, type: Object })
  private navItem!: NavItem;
}
</script>

<style lang="scss" scoped>
.nav-item {
  display: flex;
  flex-shrink: 1;
  flex-direction: row;
  width: auto;

  .link {
    display: inline-flex;
    text-decoration: none;
  }

  .text {
    display: inline-flex;
    flex-shrink: 1;
    white-space: nowrap;
    width: auto;
    font-size: medium;
    font-weight: normal;
    margin: auto 3rem auto 0.5rem;
    color: white;

    &:hover {
      color: $grey-hover;
    }
  }

  .router-link-exact-active {
    .text {
      font-weight: bold;
    }
  }

  &.disabled {
    .nav-item {
      cursor: not-allowed;
    }
    .text {
      color: $light-grey;
    }
  }

  &.primary {
    .text {
      color: $primary;
    }
  }
}
</style>
