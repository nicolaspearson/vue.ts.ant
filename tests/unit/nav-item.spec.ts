import { shallowMount } from '@vue/test-utils';

import NavItem from '@/components/NavItem.vue';

describe('NavItem.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(NavItem, {
      propsData: {
        navItem: {
          key: 1,
          text: 'Home',
          location: {
            name: 'home',
          },
          color: 'default',
        },
      },
      stubs: ['router-link'],
    });
    expect(wrapper).toMatchSnapshot();
  });
});
