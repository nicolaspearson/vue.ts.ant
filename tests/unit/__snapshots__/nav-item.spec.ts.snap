// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem.vue matches snapshot 1`] = `
<section class="nav-item default">
  <router-link-stub to="[object Object]" class="link">
    <div class="text"><span>Home</span></div>
  </router-link-stub>
</section>
`;
